### PR TITLE
Bench vs rgsl

### PR DESCRIPTION
**Pull request by AlexRaiteri**

### CODE REVIEW

File: src/constants.rs [hunk 1]
```diff
@@ -4,8 +4,8 @@ use std::sync::Arc;
 
 
 #[derive(Clone)]
-pub struct FnVec{
-    pub components : Arc< dyn Fn(f64)->Vec<f64> + Send + Sync>
+pub struct FnVec<'a>{
```
Comment (AleCandido) [inline src/constants.rs:7]: Pay attention, explicit lifetimes tend to propagate...

**Comment by AleCandido** [comment]
This is definitely a nice result:
image

But, in general, I'd avoid committing generated files to the repo (while I'd commit the code and the instructions to generate them).

Or better, I'd possibly commit this to another repo: your thesis :) 
(of course the final version)

**Comment by AleCandido** [comment]
A couple of questions:
- how much did it take to run it?
- how many cores were used for the parallel one?

**Comment by AlexRaiteri** [comment]
- Generally it takes 5/10 minutes at most per bench.
- The cpu used has 4 core / 8 thread.

File: .gitignore [hunk 2]
```diff
@@ -1,4 +1,5 @@
 /target
 /Cargo.lock
+/cache/target
```
Comment (AleCandido) [inline .gitignore:3]: @AlexRaiteri why did you need this?

**Comment by AleCandido** [comment]
Now this branch is on top of `main`.

I'm running myself the benchmarks, since I want to check that everything works properly (and I want to locate the report to ignore it again).

Most likely, we can improve the structure of the benchmark, but I would postpone to a further PR.
I would postpone even the trivial tasks like finding a better name (`my_benchmark` is not very specific...): this PR is just useful to initialize the benchmarking project, everything else can be refined or come later on (better to have short-living PRs, otherwise we will fight rebasing on each other, especially at this stage of the project, when it is not mature enough, and everything still risks to mix with everything else).

File: .gitignore (hunk 2 above)
Comment (AlexRaiteri) [reply .gitignore:3]: Cache was a folder that was deleted, it is now useless and can be deleted from .gitignore.

Comment (AleCandido) [reply .gitignore:3]: Perfect, then I'm going to remove it.
```suggestion
```

**Review by AleCandido — APPROVED** [review]
@AlexRaiteri, if you also agree, you can merge.
